tests/js/firstDiscoveryEditorTests.js

```diff
@@ -172,8 +155,128 @@ https://github.com/gpii/universal/LICENSE.txt
         jqUnit.assertEquals("The instruction text should be sourced from the active panel", expected, actual);
     };
 
-    gpii.tests.firstDiscovery.runTest("Init and navigation controls", "#gpiic-fd-navControlsTests", 1, gpii.tests.firstDiscovery.testControlss);
-    gpii.tests.firstDiscovery.runTest("Text Size", "#gpiic-fd-textSizeTests", 2, gpii.tests.firstDiscovery.testTextSize);
-    gpii.tests.firstDiscovery.runTest("TTS Hookup", "#gpiic-fd-ttsHookupTests", 2, gpii.tests.firstDiscovery.testTTSHookup);
+    gpii.tests.firstDiscovery.runTest("Init and navigation controls", "#gpiic-fd-navControlsTests", 1, gpii.tests.firstDiscovery.testControls);
+    gpii.tests.firstDiscovery.runTest("TTS Hookup", "#gpiic-fd-ttsHookupTests", 3, gpii.tests.firstDiscovery.testTTSHookup);
+
+    // Test the connection between the top level first discovery editor and the language panel: the language panel resets button positions every
+    // time when the panel itself becomes visible to accommodate the possible text or control size changes that cause the shift of button positions.
+    fluid.defaults("gpii.tests.firstDiscoveryLang", {
+        gradeNames: ["gpii.tests.firstDiscovery", "autoInit"],
+        components: {
+            prefsEditorLoader: {
+                options: {
+                    listeners: {
+                        "onPanelShown.escalate": "{firstDiscoveryLang}.events.onPanelShown"
+                    },
+                    components: {
+                        selfVoicing: {
+                            options: {
+                                // Override queueSpeech() to prevent "selfVoicing" component to continue processing to read out queued
```

Comment:
Need to JIRA this as a bug with the selfVoicing component. It should react to its own destruction by emptying its queue and nullifying its operations.